# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/CompositorWidgetChild.cpp
# Commit: d69f53c5968c
# Full Hash: d69f53c5968cf816bf07c5664b274863c155d19f
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-02-20 09:35:08
# Regressor Bug: 1604412
# File Overlap Count: 2
# Description:
#   Bug 1616000 - Fix broken push notifications r=handyman
#   
#   Bug 1604412 enabled "remote backbuffer" compositing when using non-accelerated
#   layers. However, due to my misunderstanding of how nWindow and compositing
#   handles layered windows, I broke the usecase where a child window is
# ==============================================================================

diff -r 115974724061 -r d69f53c5968c widget/windows/CompositorWidgetChild.cpp
--- a/widget/windows/CompositorWidgetChild.cpp	Wed Feb 19 23:03:12 2020 +0000
+++ b/widget/windows/CompositorWidgetChild.cpp	Wed Feb 19 22:20:33 2020 +0000
@@ -36,7 +36,8 @@
 
 bool CompositorWidgetChild::Initialize() {
   mRemoteBackbufferProvider = std::make_unique<remote_backbuffer::Provider>();
-  if (!mRemoteBackbufferProvider->Initialize(mWnd, OtherPid())) {
+  if (!mRemoteBackbufferProvider->Initialize(mWnd, OtherPid(),
+                                             mTransparencyMode)) {
     return false;
   }
 
@@ -67,6 +68,8 @@
 void CompositorWidgetChild::OnWindowModeChange(nsSizeMode aSizeMode) {}
 
 void CompositorWidgetChild::UpdateTransparency(nsTransparencyMode aMode) {
+  mTransparencyMode = aMode;
+  mRemoteBackbufferProvider->UpdateTransparencyMode(aMode);
   Unused << SendUpdateTransparency(aMode);
 }
 